From 532ae7bc976d82317211837161ac99b8e70be516 Mon Sep 17 00:00:00 2001
From: Andreas Scherer <andreas_github@freenet.de>
Date: Tue, 19 May 2026 20:16:43 +0200
Subject: [PATCH 73/73] Compile word_components with C++23.

This finishes issue #10. 'make tests demos' compiles with
C23 and C++23 flawlessly. It doesn't get better than this.
---
 PROTOTYPES/word_components.ch | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/PROTOTYPES/word_components.ch b/PROTOTYPES/word_components.ch
index 126807c..3f99ec8 100644
--- a/PROTOTYPES/word_components.ch
+++ b/PROTOTYPES/word_components.ch
@@ -10,3 +10,21 @@ main()
 @y
 int main(void)
 @z
+
+@x [2] l.48
+  for (; a; a=a->next) {@+register Vertex *u=a->tip;
+@y
+  for (; a; a=a->next) {@+Vertex *u=a->tip;
+@z
+
+@x [4] l.87
+if (u!=v->master) {@+register Vertex *w=v->master, *t;
+@y
+if (u!=v->master) {@+Vertex *w=v->master, *t;
+@z
+
+@x [5] l.116
+  if (v->master==v && v->size>1 && v->size+v->size<g->n) {@+register Vertex *u;
+@y
+  if (v->master==v && v->size>1 && v->size+v->size<g->n) {@+Vertex *u;
+@z
-- 
2.54.0

